contracts/utils/MultiRateLimited.sol

```diff
@@ -0,0 +1,273 @@
+
+    /// @notice update the ADD_MINTER_ROLE rate limit per second
+    /// @param newRateLimitPerSecond new maximum rate limit per second for add minter role
+    function updateMaxRateLimitPerSecond(uint256 newRateLimitPerSecond) external virtual override onlyGovernor {
+        require(newRateLimitPerSecond <= MAX_RATE_LIMIT_PER_SECOND, "MultiRateLimited: exceeds global max rate limit per second");
+
+        uint256 oldMaxRateLimitPerSecond = maxRateLimitPerSecond;
+        maxRateLimitPerSecond = newRateLimitPerSecond;
+
+        emit MultiMaxRateLimitPerSecondUpdate(oldMaxRateLimitPerSecond, newRateLimitPerSecond);
+    }
+
+    /// @notice update the ADD_MINTER_ROLE max buffer cap
+    /// @param newBufferCap new buffer cap for ADD_MINTER_ROLE added addresses
+    function updateMaxBufferCap(uint256 newBufferCap) external virtual override onlyGovernor {
+        require(newBufferCap <= bufferCap, "MultiRateLimited: exceeds global buffer cap");
+
+        uint256 oldBufferCap = maxBufferCap;
+        maxBufferCap = newBufferCap;
+
+        emit MultiBufferCapUpdate(oldBufferCap, newBufferCap);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the new address to add as a rateLimitedAddress
+    /// @param _rateLimitPerSecond the rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the buffer cap for this rateLimitedAddress
+    function addAddress(
+        address rateLimitedAddress,
+        uint112 _rateLimitPerSecond,
+        uint144 _bufferCap
+    ) external virtual override onlyGovernor {
+        _addAddress(rateLimitedAddress, _rateLimitPerSecond, _bufferCap);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the address whose buffer and rate limit per second will be set
+    /// @param _rateLimitPerSecond the new rate limit per second for this rateLimitedAddress
+    /// @param _bufferCap  the new buffer cap for this rateLimitedAddress
+    function updateAddress(
+        address rateLimitedAddress,
+        uint112 _rateLimitPerSecond,
+        uint144 _bufferCap
+    ) external virtual override hasAnyOfTwoRoles(TribeRoles.ADD_MINTER_ROLE, TribeRoles.GOVERNOR) {
+        if (core().hasRole(TribeRoles.ADD_MINTER_ROLE, msg.sender)) {
+            require(_rateLimitPerSecond <= maxRateLimitPerSecond, "MultiRateLimited: rate limit per second exceeds non governor allowable amount");
+            require(_bufferCap <= maxBufferCap, "MultiRateLimited: max buffer cap exceeds non governor allowable amount");
+        }
+        require(_bufferCap <= bufferCap, "MultiRateLimited: buffercap too high");
+
+        _updateAddress(rateLimitedAddress, _rateLimitPerSecond, _bufferCap);
+    }
+
+    /// @notice add an authorized rateLimitedAddress contract
+    /// @param rateLimitedAddress the new address to add as a rateLimitedAddress
+    function addAddressWithCaps(
```

Comment:
The name & docstring of this function don't represent the (powerful) feature of this - that the added address by default uses the maximum possible rate limit & buffer cap. I don't really see the need for this function, but even so, if you want to keep it, I'd suggest either using function overloading or change the name to "addAddressWithMaximumParams". Preferably the latter.